src/XrdHttp/XrdHttpReq.cc

```diff
@@ -1062,37 +1062,56 @@ void XrdHttpReq::mapXrdErrorToHttpStatus() {
+        determineXRootDChecksumFromUserDigest(userDigest, xrootdUserDigests);
+        //Get the checksums from the configuration
+        std::vector<std::string> xrootdConfigChecksums;
+        extractChecksumFromList(configChecksumList, xrootdConfigChecksums);
+        xrootdSelectedChecksum = xrootdConfigChecksums[0];
+        auto configChecksumItor = xrootdConfigChecksums.end();
+        std::find_if(xrootdUserDigests.begin(), xrootdUserDigests.end(), [&xrootdConfigChecksums, &configChecksumItor](const std::string & xrootdUserDigest){
+            configChecksumItor = std::find_if(xrootdConfigChecksums.begin(), xrootdConfigChecksums.end(), [&xrootdUserDigest](const std::string & configChecksum){
+                std::string userDigestTrimmed = xrootdUserDigest;
                 trim(userDigestTrimmed);
                 if(configChecksum.find(userDigestTrimmed) != std::string::npos) {
                     return true;
                 }
                 return false;
             });
-            return configChecksumItor != configChecksums.end();
+            return configChecksumItor != xrootdConfigChecksums.end();
         });
         //By default, the selected checksum is the first one of the configured checksum list.
-        // If the user gave a checksum that do not exist, then the checksum returned will be the default one
-        configChecksumItor = configChecksumItor != configChecksums.end() ? configChecksumItor : configChecksums.begin();
+        // If the user gave a checksum that does not exist, then the checksum returned will be the default one
+        configChecksumItor = configChecksumItor != xrootdConfigChecksums.end() ? configChecksumItor : xrootdConfigChecksums.begin();
         std::vector<std::string> checksumIdName;
         XrdOucTUtils::splitString(checksumIdName,*configChecksumItor,":");
-        selectedChecksum = checksumIdName[1];
+        xrootdSelectedChecksum = checksumIdName[1];
+    }
+}
+
+void XrdHttpReq::extractChecksumFromList(const std::string &checksumList, std::vector<std::string> &extractedChecksum) {
+    XrdOucTUtils::splitString(extractedChecksum,checksumList,",");
+}
+
+void XrdHttpReq::determineXRootDChecksumFromUserDigest(const std::string &userDigest, std::vector<std::string> &xrootdChecksums) {
+    //First, extract the user digests algorithms name
+    //Digests are separated by comas
+    std::vector<std::string> userDigestsVecWithWeights;
+    extractChecksumFromList(userDigest, userDigestsVecWithWeights);
+    //A weight can also be passed by the user for each digest with digestName;q=xx. We just discard them
+    for(auto digestWithWeight: userDigestsVecWithWeights) {
+        std::vector<std::string> currentDigest;
+        XrdOucTUtils::splitString(currentDigest, digestWithWeight, ";");
+        // Don't forget to convert the digest name coming from the user to an XRootD checksum name
+        xrootdChecksums.push_back(convert_digest_name(currentDigest[0]).c_str());
```

Comment:
I agree yes! I did not want to modify too much of the existing code. But now that I have some unit tests, I'll refactor the entire thing to extract the logic from this XrdHttpReq class.

I believe there is a need to document this mapping in the XRootD documentation. A user that will configure a sha-512 checksum (with a custom script) will not be able to call the sha-512 checksum in HTTP as the mapping will not match anything. BUT, if they call the checksum sha512 in the config, the mapping will work... I believe this needs to be documented properly to avoid surprises from the user.